### PR TITLE
Migrate jobs from ci.centos to GH actions

### PR DESCRIPTION
**Pull request by flacatus**
Signed-off-by: Flavius Lacatusu <flacatus@redhat.com>

### CODE REVIEW

**Comment by flacatus** [comment]
@AndrienkoAleksandr I change `Check nightly "opm" bundle / Check nightly "opm" bundle (pull_request)` to run only in PR.

**Comment by flacatus** [comment]
CI CENTOS jobs should fail in this PR after new changes in CI. After merge this PR centos ci jobs will be deleted

**Comment by AndrienkoAleksandr** [comment]
> @AndrienkoAleksandr I change Check nightly "opm" bundle / Check nightly "opm" bundle (pull_request) to run only in PR.

Great

**Comment by tolusha** [comment]
> Che Operator Updates / Minikube 

Testing stable versions updates / Minikube

>  Che Operator Updates / Minishift update 

Testing stable versions updates / Minishift

> Che Operator che nightly / Minishift  (pull_request) 

Testing latest changes / Minishift

>  Che Operator olm / Minikube olm changes (pull_request) 

Testing latest changes with olm / Minikube

>  Che Operator e2e tests / Minishift e2e (pull_request) 

Che Operator e2e tests / Minishift (pull_request)

**Comment by tolusha** [comment]
Could you check `ci/prow/v5-che-operator-olm-latest-changes-tests` this jobs fails ?

File: .ci/cico_olm_openshift.sh [hunk 1]
```diff
@@ -1,72 +1,100 @@
...
+#Stop execution on any error
+trap "catchFinish" EXIT SIGINT
+
+# Catch_Finish is executed after finish script.
+catchFinish() {
+  result=$?
+  if [ "$result" != "0" ]; then
+    echo "Failed on running tests. Please check logs or contact QE team (e-mail:codereadyqe-workspaces-qe@redhat.com, Slack: #che-qe-internal, Eclipse mattermost: 'Eclipse Che QE'"
+    echo "Logs should be availabe on http://artifacts.ci.centos.org/devtools/che/che-eclipse-minikube-updates/${ghprbPullId}/"
+    exit 1
```
Comment (AndrienkoAleksandr) [inline .ci/cico_olm_openshift.sh:24]: Sorry, but if you exit here, then how  will be called `getCheClusterLogs` and `archiveArtifacts` methods?

File: .ci/cico_olm_openshift.sh [hunk 2]
```diff
@@ -1,72 +1,100 @@
...
+    OPERATOR_REPO=${WORKSPACE};
+  else
+    OPERATOR_REPO=$(dirname "$SCRIPT_DIR");
+  fi
 
-# Execute olm nightly files in openshift
-PLATFORM="openshift"
-export PLATFORM
+  RAM_MEMORY=8192
+  PLATFORM="kubernetes"
```
Comment (AndrienkoAleksandr) [inline .ci/cico_olm_openshift.sh:43]: Strange, I thought that test cico_olm_openshift.sh is about openshift and should be executed on the openshift CI...

**Review by flacatus — COMMENTED** [review]
@AndrienkoAleksandr my mistake

File: .ci/cico_olm_openshift.sh (hunk 2 above)
Comment (flacatus) [reply .ci/cico_olm_openshift.sh:43]: @AndrienkoAleksandr my mistake

**Comment by flacatus** [comment]
Pending to add some secrets and add tests logs to github actions artificats

**Comment by tolusha** [comment]
Without digging into details, for the latest changes tests we have to use not the nightly version of the operator but the operator built from the PR. Do we follow this way?

**Comment by flacatus** [comment]
Finished the part to add all logs in GH artifacts: 
Example: https://github.com/eclipse/che-operator/actions/runs/266861040

File: .ci/cico_minishift_nightly.sh [hunk 3]
```diff
@@ -12,74 +12,78 @@
...
+catchFinish() {
+  result=$?
+
+  if [ "$result" != "0" ]; then
+    echo "[ERROR] Please check the artifacts in github actions"
+    getOCCheClusterLogs
+    exit 1
+  fi
+
+  echo "[INFO] JOb finished Successfully.Please check the artifacts in github actions"
```
Comment (AndrienkoAleksandr) [inline .ci/cico_minishift_nightly.sh:28]: JOb -> Job

File: .ci/cico_olm_minikube.sh [hunk 4]
```diff
@@ -43,44 +46,56 @@ export OPERATOR_IMAGE
...
+catchFinish() {
+  result=$?
+
+  if [ "$result" != "0" ]; then
+    echo "[ERROR] Please check the artifacts in github actions"
+    getCheClusterLogs
+    exit 1
+  fi
+
+  echo "[INFO] JOb finished Successfully.Please check the artifacts in github actions"
```
Comment (AndrienkoAleksandr) [inline .ci/cico_olm_minikube.sh:62]: JOb -> Job and in the other places.

File: .ci/cico_olm_minikube.sh [hunk 5]
```diff
@@ -43,44 +46,56 @@ export OPERATOR_IMAGE
...
+        sudo service docker restart
     fi
 }
 
-source "${OPERATOR_REPO}"/.ci/util/ci_common.sh
-installYQ
-installJQ
-install_VirtPackages
-# Docker should trust minikube private registry provided by "registry" addon
+echo "[INFO] Trying to get pod name of the registry proxy..."
```
Comment (AndrienkoAleksandr) [inline .ci/cico_olm_minikube.sh:98]: I guess you don't need this echo

Comment (flacatus) [reply .ci/cico_olm_minikube.sh:98]: Right, The output is assigned to a variable

File: .ci/cico_olm_minikube.sh [hunk 6]
```diff
@@ -43,44 +46,56 @@ export OPERATOR_IMAGE
...
+REGISTRY_PROXY_POD=$(kubectl get pods -n kube-system -o yaml | grep  "name: registry-proxy-" | sed -e 's;.*name: \(\);\1;') || true
```
Comment (AndrienkoAleksandr) [inline .ci/cico_olm_minikube.sh:99]: I guess, you don't use this variable below. I think the same code will be executed from startMinikube script. Github action should do that: retrieve registry pod name and await while pod is ready.

File: .ci/cico_updates_minikube.sh [hunk 7]
```diff
@@ -98,12 +91,12 @@ testUpdates() {
 
   workspaceList=$(chectl workspace:list)
   workspaceID=$(echo "$workspaceList" | grep -oP '\bworkspace.*?\b')
+  echo $workspaceID
```
Comment (AndrienkoAleksandr) [inline .ci/cico_updates_minikube.sh:94]: Seems, If we really need this log line, then it should be more informative: `echo "[INFO] Workspace id started workspace is: ${workspaceID}"`. What do you think?

Comment (flacatus) [reply .ci/cico_updates_minikube.sh:94]: +1

File: .ci/cico_olm_minikube.sh (hunk 6 above)
Comment (flacatus) [reply .ci/cico_olm_minikube.sh:99]: Right, I'll change it!

File: .ci/cico_updates_minikube.sh [hunk 8]
```diff
@@ -88,7 +81,7 @@ waitCheUpdateInstall() {
 
 testUpdates() {
   "${OPERATOR_REPO}"/olm/testUpdate.sh ${PLATFORM} ${CHANNEL} ${NAMESPACE}
-  printInfo "Successfully installed Eclipse Che previous version."
+  echo "Successfully installed Eclipse Che previous version."
```
Comment (AndrienkoAleksandr) [inline .ci/cico_updates_minikube.sh:84]: It could be a bit better:

echo "[INFO] Successfully installed Eclipse Che previous version."

Comment (flacatus) [reply .ci/cico_updates_minikube.sh:84]: +1

Comment (flacatus) [reply .ci/cico_updates_minikube.sh:84]: Also I'll use function <functionNAME> nomenclature in this file.

**Comment by flacatus** [comment]
@tolusha @AndrienkoAleksandr What do you think about adding shell check to the operator PRs? Like here: https://github.com/eclipse/che-plugin-registry/blob/master/.github/workflows/shellcheck-pr-check.yml

File: .ci/cico_updates_minishift.sh [hunk 9]
```diff
@@ -102,47 +105,50 @@ waitForNewCheVersion() {
...
-  self_signed_minishift
   installLatestCheStable
 
   # Create an workspace
   getCheAcessToken # Function from ./util/ci_common.sh
   chectl workspace:create --devfile=${OPERATOR_REPO}/.ci/util/devfile-test.yaml
 
   # Change operator images defaults in the deployment
-  sed -i -e "s|${previousPackageVersion}|${lastPackageVersion}|" "${OPERATOR_REPO}/tmp/che-operator/operator.yaml"
+  sed -i'.bak' -e "s|${previousPackageVersion}|${lastPackageVersion}|" "${OPERATOR_REPO}/tmp/che-operator/operator.yaml"
```
Comment (AndrienkoAleksandr) [inline .ci/cico_updates_minishift.sh:135]: Sorry, what does it mean  `-i'.bak'` ?

Comment (flacatus) [reply .ci/cico_updates_minishift.sh:135]: Creates a backup of your original file. In MacOS for some reason is mandatory in this cases

File: .ci/cico_updates_minishift.sh [hunk 10]
```diff
@@ -102,47 +105,50 @@ waitForNewCheVersion() {
...
+
   # Update the operator to the new release
   chectl server:update --skip-version-check --installer=operator --platform=minishift --templates="${OPERATOR_REPO}/tmp"
 
   oc patch checluster eclipse-che --type='json' -p='[{"op": "replace", "path": "/spec/server/cheImageTag", "value":"'${lastPackageVersion}'"}]' -n ${NAMESPACE}
   waitForNewCheVersion
 
   getCheAcessToken # Function from ./util/ci_common.sh
+  chectl workspace:list
```
Comment (AndrienkoAleksandr) [inline .ci/cico_updates_minishift.sh:144]: I think we could echo `workspaceList` variable below

Comment (flacatus) [reply .ci/cico_updates_minishift.sh:144]: +1

**Comment by AndrienkoAleksandr** [comment]
> @tolusha @AndrienkoAleksandr What do you think about adding shell check to the operator PRs? Like here:

+1, but it would be nice to do that in the next pr, because maybe you need to change a lot of 'bad' code to make this check working. It would prefer avoid mixing CI migration changes and shellcheck improvements.

File: .ci/cico_updates_openshift.sh [hunk 11]
```diff
@@ -28,7 +28,8 @@ init() {
   export CHANNEL="stable"
 }
 
-waitCheUpdateInstall() {
+# Utility to wait for eclipse che to be up in Openshift
```
Comment (AndrienkoAleksandr) [inline .ci/cico_updates_openshift.sh:31]: eclipse che -> Eclipse Che

File: .ci/start-minikube.sh [hunk 12]
```diff
@@ -49,11 +43,13 @@ minikube config set WantNoneDriverWarning false
 minikube config set vm-driver none
 minikube version
 
+sudo ufw disable
```
Comment (AndrienkoAleksandr) [inline .ci/start-minikube.sh:46]: Apply comment, please, why do we need this.

File: .github/workflows/olm_checks_platforms.yaml [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+    - name: Install jq
+      run: sudo pip install yq
+    - name: Install chectl
+      run: bash <(curl -sL  https://www.eclipse.org/che/chectl/) --channel=next
+    - name: Run che operator updates minikube
+      run: /bin/bash .ci/cico_olm_minikube.sh
+    - uses: actions/upload-artifact@v2
+      with:
+        name: minikube-olm-artifacts
+        path: /tmp/artifacts-che
```
Comment (AndrienkoAleksandr) [inline .github/workflows/olm_checks_platforms.yaml:32]: new line

**Review by tolusha — APPROVED** [review]

**Review by AndrienkoAleksandr — APPROVED** [review]

**Pull request merged**